api/views.py

```diff
@@ -0,0 +1,137 @@
+from rest_framework.decorators import api_view
+from rest_framework import filters, generics
+from rest_framework.response import Response
+
+from .serializers import MovieSerializer, MovieSerializerWithSchedules, HallSerializer, ScheduleSerializer, ReservationSerializer
+from cinema_project.utils import today, next_days
+from base_app.models import Movie, Cinema, Hall, Reservation, Schedule
+from . import permissions
+
+
+@api_view(['GET'])
+def currently_playing(request):
+    currently_playing_movies = Movie.objects.filter(schedules__schedule_time__range=[today(), next_days(7)])
+    serializer = MovieSerializer(currently_playing_movies, many=True)
+    return Response(serializer.data)
+
+
+@api_view(['GET'])
+def currently_playing_with_schedule(request):
+    currently_playing_movies = Movie.objects.filter(schedules__schedule_time__range=[today(), next_days(7)])
+    serializer = MovieSerializerWithSchedules(currently_playing_movies, many=True)
+    return Response(serializer.data)
+
+
+@api_view(['POST'])
+def create_hall(request):
+    cinema_name = request.data['cinema']
+    request.data['cinema'] = Cinema.objects.filter(name=cinema_name).first().pk
+    serializer = HallSerializer(data=request.data)
+    if serializer.is_valid():
+        serializer.save()
+    else:
+        print(f"Creating Hall Failed with following error: \n{serializer.errors}")
+    return Response(serializer.data)
+
+
+@api_view(['POST'])
+def update_hall(request):
+    hall_name = request.data['name']
+    cinema_name = request.data['cinema']
+    request.data['cinema'] = Cinema.objects.filter(name=cinema_name).first().pk
+    hall = Hall.objects.get(name=hall_name)
+    serializer = HallSerializer(instance=hall, data=request.data)
+
+    if serializer.is_valid():
+        serializer.save()
+    return Response(serializer.data)
+
+
+@api_view(['POST'])
+def create_movie(request):
```

Comment:
For views like this, that create resources without customizations, writing it with class based views might result in something like this:
```python
class MovieCreate(CreateAPIView):
    queryset = Movie.objects.all()
    serializer_class = MovieSerializer
```
which is easier to read and this is good for 3 reasons (that I can think of now):
- you minimize the risk of writing bad code
- the devs coming after you that are investigating an issue related to movie creation lets say, will see easily that the problem is not from within the view as it uses DRF code that is already tested and trusted.
- less code written

Reply:
Looks much cleaner!